File: languages/typescript/src/pdf.ts
import puppeteer from 'puppeteer';

export async function htmlToPdf(html: string, pdfPath: string) {
    const browser = await puppeteer.launch({
        args: ['--no-sandbox', '--disable-setuid-sandbox'],
    });
    const page = await browser.newPage();
    await page.setContent(html);
    await page.pdf({ path: pdfPath, format: 'A4' });
    await browser.close();
}

File: languages/ruby/pdf.rb
require 'puppeteer-ruby'

class Pdf
  def html_to_pdf(html, pdfFilePath)
    Puppeteer.launch(headless: true) do |browser|
      page = browser.new_page
      page.set_content(html)
      page.pdf(path: pdfFilePath, format: 'A4')
    end
  end
end